File: set4/7/gh_sol.java
import java.io.*;
import java.util.*;
class Main {
    private static final int pidMax = 1000000;
    public static void main(String []args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(bf.readLine());
        String []parse = bf.readLine().split(" ");
        int []arr = new int[t];
        int []s = new int[pidMax + 1];
        for(int i=0; i<=pidMax; i++)
            s[i] = Integer.MAX_VALUE;
        int []e = new int[pidMax + 1];
        for(int i=0; i<t; i++)
            arr[i] = Integer.parseInt(parse[i]);
        int prev = -1, ti = 1;
        for(int i=0; i<t; i++)
        {
            if(prev >= arr[i])
                ti++;
            int pid = arr[i];
            s[pid] = Math.min(s[pid], ti);
            e[pid] = Math.max(e[pid], ti);
            prev = arr[i];
        }
        int count = 0;
        StringBuilder sb = new StringBuilder();
        for(int i=0; i<=pidMax; i++){
            if(e[i] == 0)
                continue;
            count++;
            int b = e[i] - s[i] + 1;
            int c = pidMax - s[i];
            sb.append(i + " " + b + " " + c + "\n");
        }
        System.out.println(count);
        System.out.print(sb.toString());
    }
}

File: set4/7/gh_sol.py
import sys
T = int(sys.stdin.readline())
li = map(int, sys.stdin.readline().split())
s = [0x3f3f3f3f] * 1000001
e = [0] * 1000001
prev = -1
ti = 1
for cur_process in li:
    if prev >= cur_process:
        ti = ti + 1
    s[cur_process] = min(s[cur_process], ti)
    e[cur_process] = max(e[cur_process], ti)
    prev = cur_process
pn = 0
ret = [k for k in range(1000001) if e[k] != 0]
sys.stdout.write(f'{len(ret)}\n')
for i in ret:
    sys.stdout.write(f'{i} {e[i] - s[i] + 1} {1000000 - s[i]}\n')